sources/proxyspy.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+var _ = require('underscore');
+var async = require('async');
+var EventEmitter = require('events').EventEmitter || require('events');
+var request = require('request');
+
+module.exports = {
+
+	homeUrl: 'http://txt.proxyspy.net/proxy.txt',
+
+	getProxies: function(options) {
+
+		options || (options = {});
+
+		var emitter = new EventEmitter();
+
+		var fn = async.seq(
+			this.getData,
+			this.parseResponseData
+		);
+
+		fn(options, function(error, proxies) {
+
+			if (error) {
+				emitter.emit('error', error);
+			} else {
+				emitter.emit('data', proxies);
+			}
+
+			emitter.emit('end');
+		});
+
+		return emitter;
+	},
+
+	getData: function(options, cb) {
+
+		var requestOptions = {
+			method: 'GET',
+			url: 'http://txt.proxyspy.net/proxy.txt',
+		};
+
+		request(requestOptions, function(error, response, data) {
+
+			if (error) {
+				return cb(error);
+			}
+
+			if (response.statusCode >= 300) {
+				error = new Error(data);
+				error.status = response.statusCode;
+				return cb(error);
+			}
+			cb(null, { data: data, options: options });
+		});
+	},
+
+	parseResponseData: function(result, cb) {
+
+		try {
+			var data = result.data;
+			var options = result.options;
+			var countriesKeys = _.keys(options.countries);
+			var anonymityCodeMap = {
+				'n': 'transparent',
+				'a': 'anonymous',
+				'h': 'elite'
+			};
+			// regex capturing groups: IP, PORT, COUNTRY, ANONYMITYLEVEL, SECURE
+			var re = /((?:\d{1,3}\.){3}\d{1,3})\:(\d+)\s+(\S{2})\-([nah])\-*([s]*)/g;
```

Comment:
Could you use `split` and `substr` string functions to capture this information instead? I'd rather avoid regular expressions when possible.